mnemosyne/core/banks.py

```
@@ -178,15 +178,45 @@ def get_bank_stats(self, name: str) -> dict:
+        raise ValueError("Bank name cannot be empty")
+    if name == "default":
+        return  # 'default' is always valid
+    if not all(c.isalnum() or c in "-_" for c in name):
+        raise ValueError(
+            f"Invalid bank name '{name}'. Use alphanumeric, hyphens, underscores only."
+        )
+    if len(name) > 64:
+        raise ValueError(f"Bank name '{name}' exceeds 64 characters")
+
+
+def bank_exists_read_only(name: str, data_dir: Path = None) -> bool:
```

Ruff (0.15.20)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)